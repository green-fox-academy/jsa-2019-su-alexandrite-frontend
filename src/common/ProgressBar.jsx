import React from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';
import styles from './styles';

const ProgressBar = ({ progress }) => (
  <View style={styles.progressBar}>
    <View style={{ flexDirection: 'column', backgroundColor: '#4d63be', flex: progress }} />
    <View style={{ flexDirection: 'column', flex: 1 - progress }} />
  </View>
);

ProgressBar.propTypes = {
  progress: PropTypes.number.isRequired,
};

export default ProgressBar;
