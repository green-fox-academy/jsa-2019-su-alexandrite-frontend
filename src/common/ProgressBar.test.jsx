import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';
// import { useNavigation } from 'react-navigation-hooks';

import ProgressBar from './ProgressBar';

jest.mock('react-navigation-hooks');

describe('<ProgressBar />', () => {
  it('should render properly and match the snapshot', () => {
    const tree = renderer.create(
      <ProgressBar
        progress={0.5}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
