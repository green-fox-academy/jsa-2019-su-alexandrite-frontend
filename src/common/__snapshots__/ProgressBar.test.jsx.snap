// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar /> should render properly and match the snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#eee",
      "flexDirection": "row",
      "height": 4,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#4d63be",
        "flex": 0.5,
        "flexDirection": "column",
      }
    }
  />
  <View
    style={
      Object {
        "flex": 0.5,
        "flexDirection": "column",
      }
    }
  />
</View>
`;
